### DISCUSSION TITLE
Event emitting differences

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1138] [Q&A] [answered]
Good day sir, on the client side what is difference between emitting events  inside the connect event  and emitting events just on it's own[outside connect event]

**Comment by miguelgrinberg** [comment, marked as answer]
I can't think of any difference, sorry.

**Reply by Notadumdum** [reply]
So can all the event emitters and listeners be written inside the connect event on the client side?

**Reply by miguelgrinberg** [reply]
The event listeners should be defined in the global scope either as decorated functions or a class namespace, before the connection is made.

I'm not sure what you mean by "event emitters". Writing the application logic in the connect handler does not really give you any benefit over doing it right after the connect call returns.
